querier.go

```diff
@@ -253,12 +255,14 @@ func (q *blockQuerier) LabelValuesFor(string, labels.Label) ([]string, error) {
 
 func (q *blockQuerier) Close() error {
 	var merr tsdb_errors.MultiError
-
-	merr.Add(q.index.Close())
-	merr.Add(q.chunks.Close())
-	merr.Add(q.tombstones.Close())
-
-	return merr.Err()
+	if !q.closed {
```

Comment:
Usually in Golang you error out things first to have larger code blocks in "normal" flow. Can we fix that here too? Also `merr` variable should be definited just before usage.

Reply:
updated, thanks!